azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -248,10 +255,10 @@ public void run() throws Exception {
         assignUserFileOwnership(effectiveUser, getWorkingDirectory());
       }
       // Set property file permissions to <uid>:azkaban so user can write to their prop files
-      // in order to pass properties from one job to another
-      for (final File propFile : propFiles) {
+      // in order to pass properties from one job to another, except the last one
+      for (int i = 0; i < 2; i++) {
         info("Changing properties files ownership");
-        assignUserFileOwnership(effectiveUser, propFile.getAbsolutePath());
+        assignUserFileOwnership(effectiveUser, propFiles[i].getAbsolutePath());
```

Comment:
HadoopJavaJobRunnerMain is executed not as azkaban user but as _effectiveUser_. Did we test that the code reading from the jobtype private properties file in that class can actually read that file without granting permissions to it to the effectiveUser?

Reply:
I did not test that. However, I checked permissions and the user did not have permission to access the file in execution directory.